### PR TITLE
[custom_op] Change the python type that maps to ListType in schema

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #101015
* __->__ #101190
* #100980
* #100979

Previously, to specify e.g. int[], a user needed to do Tuple[int, ...].
This PR changes it to Sequence[int].

Bikeshedding: we could totally just use List[int] instead. The types
that the user gives us that we use to infer a schema is not entirely
faithful: for example, we convert `int` to SymInt.

I didn't feel strongly between Sequence[int] and List[int] so I went
with the more faithful one, plus Python recommends that you use Sequence
for input arguments (over list or tuple), though we don't subscribe to
that philosophy in general.

Test Plan:
- new test

### CODE REVIEW

File: torch/_prims/debug_prims.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def register_debug_prims():
     @custom_op("debugprims::load_tensor")
     def load_tensor(
         name: str,
-        size: Tuple[int, ...],
-        stride: Tuple[int, ...],
```
Comment (bdhirsh) [inline torch/_prims/debug_prims.py:33]: What happens if you specify `List[int]`, or `Tuple[int, ...]`. Is there a loud error? Are docs for "stuff you should know when writing a custom op schema" somewhere on the roadmap?

**Review by bdhirsh — APPROVED** [review]
nice!

Not really related to this PR, but are there any tests around passing symints into custom ops that takes `Sequence[int]` arguments?

File: torch/_prims/debug_prims.py (hunk 1 above)
Comment (zou3519) [reply torch/_prims/debug_prims.py:33]: > Is there a loud error?

There is a loud error that tells you what the accepted types are.

> Are docs for "stuff you should know when writing a custom op schema" somewhere on the roadmap?

Yeah, all of this needs a documentation page somewhere. I began one as part of the PT2 hackathon: https://docs.google.com/document/d/16YsyKY5uil_HF5mwa6UOtw0AxEHJkZp06NZcSJ3GyJs/edit#heading=h.deyt6pleeqxt , will finish after we figure out autograd support and if there should be special handling for triton

**Comment by zou3519** [comment]
> but are there any tests around passing symints into custom ops 

No, we only have tests around passing ints into custom ops that take Sequence[int]. I'll add some

**Pull request closed without merging**